src/screens/LunchScreen.js

```diff
@@ -0,0 +1,168 @@
+import PropTypes from 'prop-types';
+import React, { useCallback, useContext, useState } from 'react';
+import {
+  ActivityIndicator,
+  FlatList,
+  RefreshControl,
+  StyleSheet,
+  TouchableOpacity
+} from 'react-native';
+
+import { colors, consts, normalize, texts } from '../config';
+import {
+  BoldText,
+  Button,
+  ConnectedImagesCarousel,
+  HeaderLeft,
+  Icon,
+  LunchSection,
+  RegularText,
+  SafeAreaViewFlex,
+  Wrapper,
+  WrapperRow,
+  WrapperWithOrientation
+} from '../components';
+import { arrowLeft, arrowRight } from '../icons';
+import { useMatomoTrackScreenView, useRefreshTime } from '../hooks';
+import { graphqlFetchPolicy } from '../helpers';
+import moment from 'moment';
+import { NetworkContext } from '../NetworkProvider';
+import { useQuery } from 'react-apollo';
+import { getQuery, QUERY_TYPES } from '../queries';
+
+const { MATOMO_TRACKING } = consts;
+
+export const LunchScreen = ({ navigation }) => {
+  const [poiId, setPoiId] = useState(navigation.getParam('poiId'));
+
+  const [date, setDate] = useState(moment());
+
+  const { isConnected, isMainserverUp } = useContext(NetworkContext);
+  const refreshTime = useRefreshTime('lunch-widget', consts.REFRESH_INTERVALS.ONCE_PER_HOUR);
+  const fetchPolicy = graphqlFetchPolicy({ isConnected, isMainserverUp, refreshTime });
+
+  const currentDate = moment(date).format('YYYY-MM-DD');
+
+  const variables = {
+    dateRange: [currentDate, currentDate]
+  };
+
+  const { data, loading, refetch } = useQuery(getQuery(QUERY_TYPES.LUNCHES), {
+    fetchPolicy,
+    variables,
+    skip: !refreshTime
+  });
+
+  useMatomoTrackScreenView(MATOMO_TRACKING.SCREEN_VIEW.LUNCH);
+
+  const onPressNext = useCallback(() => {
+    setDate((oldDate) => moment(oldDate).add(1, 'day'));
+  }, [setDate]);
+
+  const onPressPrevious = useCallback(() => {
+    setDate((oldDate) => moment(oldDate).subtract(1, 'day'));
+  }, [setDate]);
+
+  const renderItem = useCallback(
+    ({ item }) => <LunchSection lunchOfferData={item} navigation={navigation} />,
+    [navigation]
+  );
+
+  let lunchData = data?.[QUERY_TYPES.LUNCHES];
+
+  if (poiId) {
+    lunchData?.filter((item) => item?.pointOfInterest?.id === poiId);
```

Comment:
re-assignment `lunchData = ` is missing here. i will add that in an upcoming PR

Reply:
you are right!
it did slip through because we only had one "lunch provider" on the internal server :(

Reply:
done in https://github.com/ikuseiGmbH/smart-village-app-app/pull/198/commits/9f909ad29054d7c93b3e0c76cbc98c30d403ddb1